packages/core/utils/src/convert-query-params.ts

```diff
@@ -411,6 +411,10 @@ const convertPopulateObject = (populate: PopulateAttributesParams, schema: Model
 };
 
 const convertNestedPopulate = (subPopulate: PopulateObjectParams, schema: Model) => {
+  if (Array.isArray(subPopulate) && _.every(subPopulate, _.isString)) {
```

Comment:
the subPopulate could be an array of path not simple string right ? `['x.y', 'ab.cd']` if so this wouldn't work

Reply:
But was it working? I thought arrays in subPopulate were not handled? :thinking: 
If arrays are accepted, we can just remove this line and let the string[] flow just like before

Reply:
Because something was broken because of that at one point during dev (tests failing), but I'll try to remove it again and see if it works

Reply:
Ha ok good to know 🤔  I think that's a nice addition anyway I was just wondering

Reply:
We might want to document that then

Reply:
Without the change, running `yarn test:api api-tests/core/upload/admin/folder.test.api.js` fails
![image](https://github.com/strapi/strapi/assets/25851739/3dc093d6-b751-4faf-ad8b-fbdcf083fc82)
But if we do except array with dot notation as subpopulate, then it could be an issue :/

What would you do?